### ISSUE TITLE
[BUG] --fatal-warnings is not working

### ISSUE THREAD

**Issue by kefasjw** [dart-code-checker/dart-code-metrics#1193]
### Environment and configuration

DCM version: 5.6.0
Dart SDK version: 2.18.6 (stable) (Tue Dec 13 21:15:14 2022 +0000) on "macos_arm64"

<details>
<summary>Configuration</summary>

```yaml
include: package:lints/recommended.yaml

analyzer:
  plugins:
    - dart_code_metrics

dart_code_metrics:
  anti-patterns:
    - long-parameter-list
  metrics:
    number-of-parameters: 3
```
</details>

### What did you do?

Run `dart run dart_code_metrics:metrics analyze lib --fatal-warnings`

```dart
void function3Param(
  int param1,
  int param2,
  int param3,
) {}

void function4Param(
  int param1,
  int param2,
  int param3,
  int param4,
) {}

```

Full code:
[reproduce_metrics.zip](https://github.com/dart-code-checker/dart-code-metrics/files/10750688/reproduce_metrics.zip)

### What did you expect to happen?

The exit code should NOT return 0

### What actually happened?

Exit code is 0

```
~/IdeaProjects/reproduce_metrics  
$ dart run dart_code_metrics:metrics analyze lib --fatal-warnings
✔ Analysis is completed. Preparing the results: 0.4s

lib/reproduce_metrics.dart:
        Long Parameter List. This function require 4 arguments.
        at /Users/xxx/IdeaProjects/reproduce_metrics/lib/reproduce_metrics.dart:7:1
        long-parameter-list : https://dcm.dev/docs/individuals/anti-patterns/long-parameter-list

~/IdeaProjects/reproduce_metrics  
$ echo $?
0
```

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

The analyzer is working in the IDE

![image](https://user-images.githubusercontent.com/11681703/219225167-0ab01345-88eb-408b-a905-5ca79a80a8e7.png)

**Comment by incendial** [comment]
Thank you for the report!

**Comment by incendial** [comment]
Available in 5.7.0 🚀
